File: Harvest/Views/Components/Chat/ChatAccent.swift
import SwiftUI

/// The three tints one chat surface needs. Field rooms are green, Seed
/// conversations are red; everything else about the two is identical, so
/// components take one of these rather than three loose colors.
struct ChatAccent {
    let base: Color     // outgoing bubble fill, send button
    let deep: Color     // outgoing gradient end
    let light: Color    // quotes, mentions, metadata

    static let rose = ChatAccent(
        base: HarvestTheme.Colors.rose,
        deep: HarvestTheme.Colors.roseDeep,
        light: HarvestTheme.Colors.accent
    )

    static let field = ChatAccent(
        base: HarvestTheme.Colors.fieldGreen,
        deep: HarvestTheme.Colors.fieldGreenDeep,
        light: HarvestTheme.Colors.fieldGreenLight
    )
}

File: android/app/src/main/java/com/harvestglass/harvest/ui/components/chat/ChatAccent.kt
package com.harvestglass.harvest.ui.components.chat

import androidx.compose.ui.graphics.Color
import com.harvestglass.harvest.ui.theme.HarvestTheme

/**
 * Port of Harvest/Views/Components/Chat/ChatAccent.swift.
 *
 * The three tints one chat surface needs. Field rooms are green, Seed
 * conversations are red; everything else about the two is identical, so
 * components take one of these rather than three loose colors.
 */
data class ChatAccent(
    /** Outgoing bubble fill, send button. */
    val base: Color,
    /** Outgoing gradient end. */
    val deep: Color,
    /** Quotes, mentions, metadata. */
    val light: Color
) {
    companion object {
        val Rose = ChatAccent(
            base = HarvestTheme.Colors.rose,
            deep = HarvestTheme.Colors.roseDeep,
            light = HarvestTheme.Colors.accent
        )

        val Field = ChatAccent(
            base = HarvestTheme.Colors.fieldGreen,
            deep = HarvestTheme.Colors.fieldGreenDeep,
            light = HarvestTheme.Colors.fieldGreenLight
        )
    }
}
